src/main/java/org/osc/controller/nsc/restserver/api/InspectionHookApis.java

```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.controller.nsc.restserver.api;
+
+import java.util.List;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.osc.controller.nsc.entities.InspectionHookEntity;
+import org.osc.sdk.controller.api.SdnControllerApi;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.jdbc.DataSourceFactory;
+import org.osgi.service.jpa.EntityManagerFactoryBuilder;
+import org.osgi.service.transaction.control.TransactionControl;
+import org.osgi.service.transaction.control.jpa.JPAEntityManagerProviderFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Component(service = InspectionHookApis.class)
+@Path("/controller/{id}/inspectionHooks")
+@Consumes({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+@Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+public class InspectionHookApis {
+
+    private static Logger LOG = LoggerFactory.getLogger(InspectionHookApis.class);
+
+    @Reference(target = "(osgi.local.enabled=true)")
+    private TransactionControl txControl;
+
+    @Reference(target = "(osgi.unit.name=nsc-mgr)")
+    private EntityManagerFactoryBuilder builder;
+
+    @Reference(target = "(osgi.jdbc.driver.class=org.h2.Driver)")
+    private DataSourceFactory jdbcFactory;
+
+    @Reference(target = "(osgi.local.enabled=true)")
+    private JPAEntityManagerProviderFactory resourceFactory;
+
+    @Reference
+    private SdnControllerApi api;
+
+    @POST
+    public String createInspectionHook(@PathParam("id") String id, InspectionHookEntity entity) throws Exception {
+
+        LOG.info(String.format("Creating inspection hook with inspection port id %s",
+                entity.getInspectionPort().getElementId()));
+
+        return RestServerApiUtils.getSdnRedirectionApi(id, this.api).installInspectionHook(entity.getInspectedPort(),
```

Comment:
i would just use the api directly. Eliminate the use of RestServerApiUtils. 
The API lifecycle/caching is managed by OSGI already. we should not duplicate that functionality.

Reply:
@arvindn05 , I have eliminated getSdnRedirectionApi from RestServerApiUtils and directly using the api's in the method. But RestServerApiUtils I would still like to retain, as the HASH updates between Port, inspectionPort and inspectionHook are managed in this utility along with the id validatin method.

Reply:
not sure i understand this comment. i dont see any id validation either.

It just seems like you are caching the sdn redirection api in a static map.
if you are caching the api's, who is clearing the map? How do we know the api in the map is still valid? the DB connection can close underneath...

Reply:
@arvindn05,yes each api hold entity manager handle, which is used to access the db. The cache entity is not cleared, good point. I am thinking of flushing this cache entries as part of 'close' method of networkapis or other option is when the Ser let is getting destroyed or any other option?
Regarding Id validation it can be any ip address and I can check for null value as part of validation.